lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -205,11 +213,33 @@ public Builder setDimensions(DimensionList dimensions) {
      * (Optional) Set the timestamp for the exported metric line. In most cases, {@link
      * Builder#setCurrentTime()} should be suitable.
      *
+     * <p>If the timestamp is from before the year 2000 or from after the year 3000 (e.g., when the
+     * wrong unit was used when creating the {@link Instant}), the timestamp will be discarded and
+     * no value will be set.
+     *
      * @param timestamp an {@link Instant} object describing the time at which the {@link Metric}
      *     was created.
      * @return this
      */
     public Builder setTimestamp(Instant timestamp) {
+      int year = timestamp.atZone(ZoneOffset.UTC).getYear();
+      if (year < 2000 || year > 3000) {
+        if (timestampWarningCounter.getAndIncrement() == 0) {
+          logger.warning(
+              String.format(
+                  "Order of magnitude of the timestamp seems off (%s). "
+                      + "The timestamp represents a time before the year 2000 or after the year 3000. "
+                      + "Skipping setting timestamp, the current server time will be added upon ingestion. "
+                      + "Only one out of every %d of these messages will be printed.",
+                  timestamp.toString(), TIMESTAMP_WARNING_THROTTLE_FACTOR));
+        }
+        timestampWarningCounter.compareAndSet(TIMESTAMP_WARNING_THROTTLE_FACTOR, 0);
+
+        // do not set the timestamp, metric will be exported without timestamp and the current
+        // server timestamp is added upon ingestion.
+        return this;
+      }
+
       this.time = timestamp;
```

Comment:
This was able to deal with null values before since it would just set null on the property but now it would throw. Please wrap it in a null check.